Sources/TuistSupport/Utils/GitHandler.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+import TSCBasic
+
+public protocol GitHandling {
+    /// Clones the given `url` **into** the given `path`.
+    /// `path` must point to a directory where a git repo can be cloned.
+    ///
+    /// - Parameters:
+    ///   - url: The `url` to the git repository to clone.
+    ///   - path: The `AbsolutePath` to clone the git repository.
+    func clone(url: String, into path: AbsolutePath) throws
+
+    /// Clones the given `url` **to** the given `path`.
+    /// `path` must point to a directory where a git repo can be cloned.
+    ///
+    /// - Parameters:
+    ///   - url: The `url` to the git repository to clone.
+    ///   - path: The `AbsolutePath` to clone the git repository.
+    func clone(url: String, to path: AbsolutePath?) throws
+
+    /// Checkout to some git `id` in the given `path`.
+    ///
+    /// The `id` must be something known to the `git checkout` command, which includes:
+    ///  - A branch, i.e. `main`.
+    ///  - A tag, i.e. `1.0.0`
+    ///  - A sha, i.e. `028c13b`
+    ///
+    /// - Parameters:
+    ///   - id: An identifier for the `git checkout` command.
+    ///   - path: The path to the git repository (location with `.git` directory) in which to perform the checkout.
+    func checkout(id: String, in path: AbsolutePath?) throws
+}
+
+/// An implementation of `GitHandling`.
+/// Uses the system to execute git commands. 
+public final class GitHandler: GitHandling {
+    private let system: Systeming
+
+    public init(
+        system: Systeming = System.shared
+    ) {
+        self.system = system
+    }
+
+    public func clone(url: String, into path: AbsolutePath) throws {
```

Comment:
I'd suggest making these methods asynchronous using a Reactive interface. You can either use the existing Rxswift-based methods or extend `System` to use Combine. The idea is to slowly get rid of RxSwift in favor of combine so I think this can be a great first step. Note that the project has [this library](https://github.com/CombineCommunity/CombineExt) as a dependency that comes with some syntactic-sugar and useful operators.

Reply:
I forgot to mention that the motivation for being able to run clones asynchronously is that we can run multiple git commands simultaneously, for example doing multiple clones :)

Reply:
Actually https://github.com/tuist/tuist/pull/2108

Reply:
@pepibumur Nice! I'll update to use `Combine` soon!